test/test_jit.py

```diff
@@ -14210,31 +13986,6 @@ def forward(self, x):
         weak_mod.weight = torch.nn.Parameter(torch.ones(5, 5) * 100)
         self.assertFalse(strong_mod(inp).allclose(weak_mod(inp)))
 
-    @unittest.skipIf(hasattr(torch.jit, 'WeakScriptModuleProxy'), "# TODO: re-enable"
-                                                                  "this when WeakScriptModuleProxy has been deleted")
-    def test_weak_module_isinstance(self):
```

Comment:
This one we should probably keep

Reply:
it's not testing anything useful, the `isinstance` is functionally broken now. It only works in the body because `self.linear` hasn't been scripted yet since the recursive script call happens after `__init__`.
Example:
tester = self
```
class M(torch.jit.ScriptModule):
    def __init__(self):
        super(M, self).__init__()
        self.linear = nn.Linear(2, 2)
        tester.assertTrue(isinstance(self.linear, nn.Linear))

mod = M()
self.assertFalse(isinstance(mod.linear, nn.Linear))
```